File: Giscup2016Java/src/giscup/lineparsers/LineParser.java
package giscup.lineparsers;

public interface LineParser {
	/**
	 * 
	 * Parse a line
	 * 
	 * @param xyt result vector
	 * @param line line to be parsed
	 * @return line contains a valid/invalid record
	 * 
	 */
	
	boolean parseLine(int[] xyt, String line);
	
}

File: Giscup2016Spark/src/giscup/parser/LineParser.scala
package giscup.parser
/**
 * 
 * Line Parser trait
 * 
 * it only has one function which calculates the event's XYT parameters only if the line contains a valid event 
 * 
 */
trait LineParser {
  def parseLine(xyt: Array[Int], line: String): Boolean
}